inference-engine/src/readers/onnx_reader/onnx_model_validator.hpp

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <istream>
+
+namespace InferenceEngine {
+    // 2 empty structs used for tag dispatch below
+    struct onnx_format {};
+    struct prototxt_format {};
+
+    bool is_valid_model(std::istream& model, onnx_format);
+
+    bool is_valid_model(std::istream& model, prototxt_format);
```

Comment:
Why do we need to have two functions? Maybe we can pass extension and inside the method have a different branches for different formats.

Reply:
This is because we have a helper `readPathFromStream` in `ie_onnx_reader.cpp` and I wanted to use it to avoid code duplication. This way the branching is done already in `ONNXReader::supportModel` and not in `is_valid_model`.